File: string-paradigm-expression_2/src/main/java/org/jesperancinha/string/paradigm/original/expression/RegionSpellingCheck.java
package org.jesperancinha.string.paradigm.original.expression;

import java.util.ArrayList;
import java.util.List;

public class RegionSpellingCheck extends SpellingCheck {

	private final List<RegionImpl> spellingDoubleCommaRegions = new ArrayList<RegionImpl>();
	private final List<RegionImpl> spellingWhiteSpacesRegions = new ArrayList<RegionImpl>();
	private final List<RegionImpl> spellingBeginCapitalRegions = new ArrayList<RegionImpl>();

	public void checkSpelling(RegionImpl component) {
		if (checkDoubleCommas(component.getDescription())) {
			spellingDoubleCommaRegions.add(component);
		}
		if (checkWhiteSpaces(component.getDescription())) {
			spellingWhiteSpacesRegions.add(component);
		}
		String[] elements = component.getDescription().split(",");
		for (int i = 0; i < elements.length; i++) {
			if (checkCapitalLetter(elements, i)) {
				spellingBeginCapitalRegions.add(component);
			}
		}
	}

	public List<RegionImpl> getSpellingDoubleCommaRegions() {
		return spellingDoubleCommaRegions;
	}

	public List<RegionImpl> getSpellingWhiteSpacesRegions() {
		return spellingWhiteSpacesRegions;
	}

	public List<RegionImpl> getSpellingBeginCapitalRegions() {
		return spellingBeginCapitalRegions;
	}
}

File: jeorg-kotlin-apps/string-array-paradigms/string-paradigm-expression_2/src/main/kotlin/org/jesperancinha/string/paradigm/original/expression/RegionSpellingCheck.kt
package org.jesperancinha.string.paradigm.original.expression

class RegionSpellingCheck : SpellingCheck() {
    val spellingDoubleCommaRegions: MutableList<RegionImpl> = ArrayList()
    val spellingWhiteSpacesRegions: MutableList<RegionImpl> = ArrayList()
    val spellingBeginCapitalRegions: MutableList<RegionImpl> = ArrayList()
    fun checkSpelling(component: RegionImpl) {
        if (checkDoubleCommas(component.description)) {
            spellingDoubleCommaRegions.add(component)
        }
        if (checkWhiteSpaces(component.description)) {
            spellingWhiteSpacesRegions.add(component)
        }
        val elements: Array<String> = component.description.split(",".toRegex()).dropLastWhile { it.isEmpty() }
            .toTypedArray()
        for (i in elements.indices) {
            if (checkCapitalLetter(elements, i)) {
                spellingBeginCapitalRegions.add(component)
            }
        }
    }
}